support-frontend/assets/helpers/redux/checkout/payment/directDebit/thunks.ts

```diff
@@ -0,0 +1,81 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { checkAccount } from 'components/directDebit/helpers/ajax';
+import type { PaymentAuthorisation } from 'helpers/forms/paymentIntegrations/readerRevenueApis';
+import { DirectDebit } from 'helpers/forms/paymentMethods';
+import type { ContributionsState } from 'helpers/redux/contributionsStore';
+import type { SubscriptionsState } from 'helpers/redux/subscriptionsStore';
+
+export const directDebitErrorMessages = {
+	notConfirmed: 'You need to confirm that you are the account holder.',
+	notCompletedRecaptcha: "Please check the 'I'm not a robot' checkbox",
+	invalidInput:
+		'Your bank details are invalid. Please check them and try again',
+	incorrectInput:
+		'Your bank details are not correct. Please check them and try again',
+	default: 'Oops, something went wrong, please try again later',
+};
+
+type DirectDebitConfirmationResponse = {
+	accountValid: boolean;
+};
+
+export const confirmAccountDetails = createAsyncThunk<
+	boolean,
+	void,
+	{
+		state: SubscriptionsState | ContributionsState;
+	}
+>(
+	'directDebit/confirmAccountDetails',
+	async function confirmAccountDetails(_, { getState }) {
+		const {
+			sortCodeString,
+			sortCodeArray,
+			accountNumber,
+			accountHolderConfirmation,
+		} = getState().page.checkoutForm.payment.directDebit;
+		const sortCode = sortCodeArray.join('') || sortCodeString;
+		const isTestUser = getState().page.user.isTestUser ?? false;
+		const { csrf } = getState().page.checkoutForm;
+
+		if (!accountHolderConfirmation) {
+			throw new Error(directDebitErrorMessages.notConfirmed);
+		}
+
+		const response = await checkAccount(
+			sortCode,
+			accountNumber,
+			isTestUser,
+			csrf,
+		);
+		if (!response.ok) {
+			throw new Error(directDebitErrorMessages.invalidInput);
+		}
+		const checkAccountStatus =
+			(await response.json()) as DirectDebitConfirmationResponse;
+
+		return checkAccountStatus.accountValid;
+	},
+);
+
+export const payWithDirectDebit = createAsyncThunk<
+	void,
+	(authorisation: PaymentAuthorisation) => void,
+	{
+		state: SubscriptionsState | ContributionsState;
+		rejectValue: string;
+	}
+>(
+	'directDebit/payWithDirectDebit',
+	function payWithDirectDebit(onPaymentAuthorisation, { getState }) {
```

Comment:
Currently, the contribs page assumes this method will validate if the recaptcha has been clicked (https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/components/directDebit/directDebitActions.ts#L193). That means right now this change will let you submit the direct debit on contribs without checking the recaptcha.

It's slightly different on the subs direct debit, where the component does that check. We could maybe do that for contribs too? Or move the check here and remove it from the subs component? Either way, seems like a good opportunity to unify some code!

Reply:
Ah, I missed that change to the old function. I'll take a look at this.

Reply:
I'm going to move the check inside the component too- the subs checkout doesn't actually use this function, so it wouldn't be helpful to use it here.